### PR TITLE
[WIP][Live] Let the bidding button know about being bidder registration

### PR DESCRIPTION
**Pull request by orta**
This lets the bid button know about the registration status of the user. We don't show anything different if you're a logged in user who ins't registered ( I initially created that state, but went as saw all examples ( and past code) were the same )

Note: This PR makes us _ignore_ events that we don't understand, I'm a little bit torn there, but it's either that or we whitelist events that aren't relevant to us to show e,g, `StartingPriceChanged`/`ReservePriceChanged`

WIP cause tests, and cause I want to try get the "you're the max bidder" in before I make a beta, giving myself 30m

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
        
            1 Error
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">No CHANGELOG changes made</td>
    </tr>
    
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            3 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">PR is classed as Work in Progress</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |258.6ms | [AuctionViewController.swift](/artsy/eigen/tree/orta-live_bidding/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
197.2ms | [SwiftyJSON.swift](/artsy/eigen/tree/orta-live_bidding/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
173.2ms | [WebSocket.swift](/artsy/eigen/tree/orta-live_bidding/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
65.5ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/orta-live_bidding/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift [hunk 1]
```diff
@@ -121,7 +121,9 @@ private extension PrivateFunctions {
 
         let existingEventIds = Set(lot.eventIDs)
         let newEvents = lotEvents.filter { existingEventIds.contains($0["eventId"] as? String ?? "") == false }
-        lot.addEvents( newEvents.map { LiveEvent(JSON: $0) } )
+
+        // TODO: is this a good idea? This will remove events we don't know yet
+        lot.addEvents( newEvents.flatMap { LiveEvent(JSON: $0) } )
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: @ashfurrow opinions?

Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: Good question. I've been wondering this too. I think getting input from @acjay would be a good idea. Prediction currently shows "wtf" for unknown events, not sure if that will ship though 😉

Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: @acjay recommended it to me to go with ignoring, but I'm very open to ideas here

Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: Maybe we can set up a flag to ignore for regular users but show for artsy users? Make it easier to discover and debug these things?

Comment (acjay) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: If there's a "debug mode" or something, that might a nice switch to have. I think staging modes should as much as possible be WYSIWYG for production, but maybe with escape hatches to get debug behavior. On web, it's easy to do this with a query string, but I guess that's not so easy on phone.

**Comment by ashfurrow** [comment]
I'm a huge fan of this PR, great work @orta.

**Comment by orta** [comment]
Cool, merging.

File: Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift (hunk 1 above)
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: We can detect if it's an admin vs a non-admin and show them?

Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: Yeah, we could use `[ARAppStatus isBetaDevOrAdmin]`?

**Pull request merged**